### PR TITLE
refactored code as per SOLID

### PR DESCRIPTION
**Pull request by ShridharDaga**
refactored code as per SOLID

### CODE REVIEW

File: java/src/main/java/com/codurance/training/tasks/TaskAdd.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+import java.io.PrintWriter;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+
+public class TaskAdd {
```
Comment (AbhilashZemoso) [inline java/src/main/java/com/codurance/training/tasks/TaskAdd.java]: add an interface for this class

File: java/src/main/java/com/codurance/training/tasks/TaskAdd.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            date = new SimpleDateFormat("dd-MM-yyyy").parse(deadline);
+        }
+        catch (ParseException e) {
+            e.printStackTrace();
+        }
+
+        return date;
+    }
+
+    public long nextId() {
```
Comment (AbhilashZemoso) [inline java/src/main/java/com/codurance/training/tasks/TaskAdd.java]: make this private

**Review by AbhilashZemoso — CHANGES_REQUESTED** [review]
work on comments

File: java/src/main/java/com/codurance/training/tasks/TaskAdd.java (hunk 1 above)
Comment (ShridharDaga) [reply java/src/main/java/com/codurance/training/tasks/TaskAdd.java]: addressed

File: java/src/main/java/com/codurance/training/tasks/TaskAdd.java (hunk 2 above)
Comment (ShridharDaga) [reply java/src/main/java/com/codurance/training/tasks/TaskAdd.java]: addressed

**Review by AbhilashZemoso — APPROVED** [review]
LGTM

**Pull request merged**